src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.example.bookshop.service.impl;
+
+import com.example.bookshop.dto.BookDto;
+import com.example.bookshop.dto.CreateBookRequestDto;
+import com.example.bookshop.exception.EntityNotFoundException;
+import com.example.bookshop.mapper.BookMapper;
+import com.example.bookshop.model.Book;
+import com.example.bookshop.repository.BookRepository;
+import com.example.bookshop.service.BookService;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.IntStream;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private static final String ISBN_START = "978";
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private StringBuilder stringBuilder;
+    private Random random;
+
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+        random = new Random();
+    }
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn(generateIsbn());

```

Comment:
It is overcodding. Why do you create random isbn? you should pass it in CreateBookRequestDto because you have a bookstore not a printery ))